tests/core/model_management/test_local_model_trainer.py

```diff
@@ -0,0 +1,209 @@
+
+
+def test_train_save_and_load(trainer_type_cfg, save_path):
+    """Test that a trained model can be loaded"""
+    trainer = local_trainer(**trainer_type_cfg)
+    model_future = trainer.train(
+        SampleModule,
+        training_data=DataStream.from_iterable([]),
+        save_path=save_path,
+    )
+    model_future.wait()
+
+    # Make sure it can be loaded manually
+    model = SampleModule.load(save_path)
+    assert isinstance(model, SampleModule)
+
+    # Make sure that it can be loaded via the future
+    # NOTE: With a subprocess, this requires that save path is given so that the
+    #   model can be re-loaded from disk
+    model = model_future.load()
+    assert isinstance(model, SampleModule)
+
+
+def test_save_with_id(trainer_type_cfg, save_path):
+    """Test that saving with the training id correctly injects the ID in the
+    save path
+    """
+    trainer = local_trainer(**trainer_type_cfg)
+    model_future = trainer.train(
+        SampleModule,
+        training_data=DataStream.from_iterable([]),
+        save_path=save_path,
+        save_with_id=True,
+    )
+    model_future.wait()
+    assert model_future.save_path != save_path
+    assert model_future.id in model_future.save_path
+    assert os.path.exists(model_future.save_path)
+
+
+def test_cancel_clean_termination(trainer_type_cfg):
+    """Test that cancelling an in-progress training successfully destroys the
+    training when the training is run in a way that can be shut down cleanly
+    """
+    trainer = local_trainer(**trainer_type_cfg)
+    model_future = trainer.train(
+        SampleModule,
+        training_data=DataStream.from_iterable([]),
+        sleep_time=1000,
+    )
+    assert model_future.get_status() == ModelTrainerBase.TrainingStatus.RUNNING
+    assert not model_future.get_status().is_terminal
+
+    # Cancel the future
+    model_future.cancel()
+    assert model_future.get_status() == ModelTrainerBase.TrainingStatus.CANCELED
+    assert model_future.get_status().is_terminal
+    model_future.wait()
+
+
+def test_cancel_without_waiting(trainer_type_cfg):
+    """Test that cancelling an in-progress training that uses a long sleep (and
+    can't be easily destroyed in a thread) still reports CANCELED as the status
+    before the training has fully terminated.
+    """
+    trainer = local_trainer(**trainer_type_cfg)
+    model_future = trainer.train(
+        SampleModule,
+        training_data=DataStream.from_iterable([]),
+        sleep_time=0.5,
+        sleep_increment=0.5,
+    )
+    assert model_future.get_status() == ModelTrainerBase.TrainingStatus.RUNNING
+    assert not model_future.get_status().is_terminal
+
+    # Cancel the future and make sure it reports canceled, even though the
+    # function is still sleeping
+    model_future.cancel()
+    assert model_future.get_status() == ModelTrainerBase.TrainingStatus.CANCELED
+    assert model_future.get_status().is_terminal
+    model_future.wait()
```

Comment:
Thoughts on not doing `model_future.wait()` here?
This will add a full second to the test runs, and I had just gone and removed all of these sleepy cases from the old training tests to take the suite from 30s to 5s.

If it's possible to add a global teardown fixture that can `.wait()` on a big old list of arbitrary futures that could be nice, I really just don't want this to have to happen serially here and bog everything down

Reply:
I'm fine with that. It will wrap up in the background